lib/std/net.zig

```diff
@@ -839,6 +836,20 @@ fn linuxLookupName(
             if (addrs.items.len == 0) {
                 try linuxLookupNameFromDnsSearch(addrs, canon, name, family, port);
             }
+            if (addrs.items.len == 0) {
+                // RFC 6761 Section 6.3
+                // Name resolution APIs and libraries SHOULD recognize localhost
+                // names as special and SHOULD always return the IP loopback address
+                // for address queries and negative responses for all other query
+                // types.
+
+                // Check for equal to "localhost" or ends in ".localhost"
+                if (mem.endsWith(u8, name, "localhost") and (name.len == "localhost".len or name[name.len - "localhost".len] == '.')) {
```

Comment:
Shouldn't the check be non case-sensitive?

Reply:
The other functions (e.g. `linuxLookupNameFromHosts`) are also case sensitive: I think case normalisation is meant to happen before `linuxLookupName` is called.